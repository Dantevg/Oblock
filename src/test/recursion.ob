fac1 n => {
	if n == 0: return 1
	else return n * fac1(n-1)
}

fac2 n => n == 0 and 1 or n * fac2(n-1)

fac3 n => match n [
	0 => 1
	n => n * fac3(n-1)
]

fac4 = case [
	0 => 1
	n => n * fac3(n-1)
]

print(fac1 5)
print(fac2 5)
print(fac3 5)
print(fac4 5)
