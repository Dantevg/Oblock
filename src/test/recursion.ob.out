120
120
120
